homeassistant/components/advantage_air/__init__.py

```diff
@@ -50,19 +51,25 @@ async def async_get():
         update_interval=timedelta(seconds=ADVANTAGE_AIR_SYNC_INTERVAL),
     )
 
-    async def async_change(change):
-        try:
-            if await api.async_change(change):
-                await coordinator.async_refresh()
-        except ApiError as err:
-            _LOGGER.warning(err)
+    def error_handle_factory(func):
+        """Return the provided API function wrapped in an error handler and coordinator refresh."""
+
+        async def error_handle(param):
+            try:
+                if await func(param):
+                    await coordinator.async_refresh()
```

Comment:
```suggestion
                    await coordinator.async_request_refresh()
```

Reply:
@bdraco does this change the behavior of pulling refreshes? My Tests are now failing like the refresh is not being done.

Reply:
> Refresh will wait a bit to see if it can batch them.
Right, ill update tests

Reply:
Yes, it has a debouncer so you don't overload the device

https://github.com/home-assistant/core/blob/37395ecd36fb75792a093289c25a44366e3d66af/homeassistant/helpers/update_coordinator.py#L153

You may need to call `async_fire_time_changed` or lower the debounce time on the coordinator